{{ cookiecutter.project_slug }}/{{ cookiecutter.package_name }}/strategies/download.py

```diff
@@ -29,66 +30,70 @@ class FileConfig(BaseModel):
             "Encoding used when opening the file. The default is platform dependent."
         ),
     )
+    datacache_config: Optional[DataCacheConfig] = Field(
+        None,
+        description=(
+            "Configurations for the data cache for storing the downloaded file "
+            "content."
+        ),
+    )
 
 
-@dataclass
-class DemoFileStrategy:
-    """Strategy for retrieving data via local file."""
+class FileResourceConfig(ResourceConfig):
+    """File download strategy filter config."""
 
-    download_config: "ResourceConfig"
+    downloadUrl: FileUrl = Field(  # type: ignore[assignment]
+        ..., description="The file URL, which will be downloaded."
+    )
+    configuration: FileConfig = Field(
+        FileConfig(), description="File download strategy-specific configuration."
+    )
 
-    def initialize(
-        self, session: "Optional[Dict[str, Any]]" = None
-    ) -> "Dict[str, Any]":
-        """Initialize strategy.
+    @validator("downloadUrl")
+    def ensure_path_exists(cls, value: FileUrl) -> FileUrl:
+        """Ensure `path` is defined in `downloadUrl`."""
+        if not value.path:
+            raise ValueError("downloadUrl must contain a `path` part.")
+        return value
 
-        This method will be called through the `/initialize` endpoint of the OTE-API
-        Services.
 
-        Parameters:
-            session: A session-specific dictionary context.
+class SessionUpdateFile(SessionUpdate):
+    """Class for returning values from Download File strategy."""
 
-        Returns:
-            Dictionary of key/value-pairs to be stored in the sessions-specific
-            dictionary context.
+    key: str = Field(..., description="Key to access the data in the cache.")
 
-        """
-        return {}
 
-    def get(self, session: "Optional[Dict[str, Any]]" = None) -> "Dict[str, Any]":
-        """Execute the strategy.
+@dataclass
+class FileStrategy:
+    """Strategy for retrieving data from a local file.
 
-        This method will be called through the strategy-specific endpoint of the
-        OTE-API Services.
+    **Registers strategies**:
 
-        Parameters:
-            session: A session-specific dictionary context.
+    - `("scheme", "fileDEMO")`
 
-        Returns:
-            Dictionary of key/value-pairs to be stored in the sessions-specific
-            dictionary context.
+    """
 
-        """
-        if (
-            self.download_config.downloadUrl is None
-            or self.download_config.downloadUrl.scheme != "file"
-        ):
-            raise ValueError(
-                "Expected 'downloadUrl' to have scheme 'file' in the configuration."
-            )
-        filename = Path(self.download_config.downloadUrl.host).resolve()
+    download_config: FileResourceConfig
```

Comment:
Wouldn't file_config be a more natural name?

Reply:
No, It's not allowed. The interface states that this attribute *MUST* be `download_config`.